client/log_verifier.go

```diff
@@ -132,9 +133,11 @@ func (c *logVerifier) BuildLeaf(data []byte) (*trillian.LogLeaf, error) {
 	if err != nil {
 		return nil, err
 	}
+	h := sha256.New()
+	h.Write(data)
 	return &trillian.LogLeaf{
 		LeafValue:        data,
 		MerkleLeafHash:   leafHash,
-		LeafIdentityHash: leafHash,
+		LeafIdentityHash: h.Sum(nil),
```

Comment:
Since this is completely bogus, rather than putting something which looks plausible on the surface could we have a const which we assign here, preferably a string like "INVALID" or something?

Reply:
Well it's not completely bogus... If two leafs have the same identity hash, the second one won't get stored.  The hash of the leaf data is a perfectly sane choice. 

The only thing that seems to be missing here is the ability to swap out the sha256 algorithm for another one which I think we should do as soon as we have a use case for it. wdyt?

Reply:
Blindly setting the `LeafIdentityHash` to the hash of the leaf data is not a one-size-fits-all choice: CT, for example, explicitly doesn't do that (it uses `H(cert)`, not `H(leaf)`) and you'd open the logs to spam if the CTFE personality uses this library.

`LeafIdentityHash` is only be used by the log when submitting entries, and since Trillian is the layer below the personality you need a way of having the personality which is using this library to tell it what the ID hash of a particular leaf is.